pos_frequency_pipeline.py

```diff
@@ -1,8 +1,69 @@
 """
 Implementation of POSFrequencyPipeline for score ten only.
 """
+import json
+import re
+from collections import Counter
+from pathlib import Path
+
+from pymystem3 import Mystem
+
+from article import Article
+from constants import ASSETS_PATH
+from pipeline import CorpusManager
+from visualizer import visualize
 
 
 class POSFrequencyPipeline:
-    def __init__(self, assets):
-        pass
+    def __init__(self, assets: CorpusManager):
+        self.assets = assets
+        self.text = ""
+        self.frequencies = Counter()
+
+    def run(self):
+        for idx in self.assets.get_articles():
+            article = Article(url=None, article_id=idx)
+            self.text = article.get_raw_text()
+            self.frequencies += self._process()
+            self._update_meta(idx)
+            visualize(
+                statistics=self.frequencies,
+                path_to_save=Path(ASSETS_PATH) / f"{idx}_image.png",
+            )
+
+    def sample(self):
+        raise NotImplementedError
+
+    def _process(self):
```

Comment:
the pipeline does not analyze the text - it works on top of already processed files. change the logic accordingly

Reply:
addressed